### PR TITLE
Refactored mutli_class_road_assignment

### PR DESCRIPTION
**Pull request by amitsandhel**
removed the nested library and fixed the print statements to be both py2 and py3 compatabile. Also fixed the nested statement to use multiple with statements

### CODE REVIEW

File: TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py [hunk 1]
```diff
@@ -512,8 +518,8 @@ def get_attribute_name(at):
                                 else:
                                     val = 'undefined'
                             
-                                print "Primary assignment complete at %s iterations." %number
-                                print "Stopping criterion was %s with a value of %s." %(stoppingCriterion, val)
+                                print ("Primary assignment complete at %s iterations.") %number
+                                print ("Stopping criterion was %s with a value of %s.") %(stoppingCriterion, val)
```
Comment (JamesVaughan) [inline TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py]: The format string needs to be inside of the brackets.

File: TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py [hunk 2]
```diff
@@ -249,7 +258,7 @@ def __call__(self, xtmf_ScenarioNumber, Mode_List, xtmf_Demand_String, TimesMatr
                 if str(demandMatrix).lower() == 'mf0':
                     dm = _util.initializeMatrix(matrix_type='FULL')
                     demandMatrix = dm.id
-                    print "Assigning a Zero Demand matrix for class '%s' on scenario %d" %(str(self.ClassNames[i]),int(self.Scenario.number))
+                    print ("Assigning a Zero Demand matrix for class '%s' on scenario %d") %(str(self.ClassNames[i]),int(self.Scenario.number))
```
Comment (JamesVaughan) [inline TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py]: The format string needs to be inside of the brackets.

File: TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py [hunk 3]
```diff
@@ -33,17 +34,25 @@
...
 import inro.modeller as _m
 import traceback as _traceback
 import multiprocessing
 from contextlib import contextmanager
-from contextlib import nested
+import six
+if six.PY3:
+    _m.InstanceType = object
+    _m.TupleType = object
+    _m.ListType = object
```
Comment (JamesVaughan) [inline TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py]: Even if we are not using the list type call
```Python
 _util.initalizeModellerTypes(_m).
```

**Review by JamesVaughan — CHANGES_REQUESTED** [review]

File: TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py (hunk 1 above)
Comment (amitsandhel) [reply TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py]: fixed the issue

File: TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py (hunk 2 above)
Comment (amitsandhel) [reply TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py]: fixed

File: TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py (hunk 3 above)
Comment (amitsandhel) [reply TMGToolbox/src/XTMF_internal/multi_class_road_assignment.py]: fixed

**Pull request merged**